src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvoker.cs

```diff
@@ -672,12 +690,15 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
 
                         if (exceptionContext != null)
                         {
-                            if (exceptionContext.Result != null && !exceptionContext.ExceptionHandled)
+                            if (exceptionContext.Result != null ||
+                                exceptionContext.Exception == null ||
+                                exceptionContext.ExceptionHandled)
```

Comment:
It's possible the table we came up with still has a bug. In MVC 5.x, setting the `Result` alone changes nothing because the `Exception` will still be rethrown. I believe that was also the case in ASP.NET Core MVC 1.0.0. Should I double-check with 1.0.0?

If I'm wrong, this is fine for the patch release. We should revisit the `exceptionContext.Result != null && exceptionContext.Exception != null && !exceptionContext.ExceptionHandled` behaviour for 2.0. It doesn't make sense (and confused our customers) to have the `Result` used when they haven't indicated the `Exception` should be suppressed.